browser/src/Plugins/Api/Process.ts

```diff
@@ -27,7 +28,8 @@ const mergeSpawnOptions = (originalSpawnOptions: ChildProcess.ExecOptions | Chil
         },
     }
 
-    const existingPath = process.env.Path || process.env.PATH
+    const shellEnvironment = shellEnv.sync()
```

Comment:
Is it possible to use the `async` version of the API? I'm on a crusade to try and minimize `sync` calls since they can sometimes cause stutters in the UX. 

In this case, this would be called when opening up a file, and there are a few other bottlenecks we have at that point.

I'm thinking of a couple alternatives:
- We could switch to using the async version here, but that might be painful, because we'd need to wire up `async` and `Promise` up to the call sites
- We could call the sync version once on startup. I'm looking at how we can streamline our startup (#938), and we could potentially switch to using the async version at that point, and execute in parallel to some of our other startup tasks. But for now, we could just call the sync version on startup to seed the values. It shouldn't change while we're running, AFAIK

Reply:
I opted for option one here and used async await and added returned promise typings up to the call sites hopefully this doesn't clash with #938